app.rb

```diff
@@ -0,0 +1,189 @@
+      puts 'Person created successfully'
+    else
+      puts 'Invalid option'
+    end
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Classroom:'
+    classroom = Classroom.new(gets.chomp)
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    if parent_permission == 'Y' || parent_permission.downcase == 'y'
+      parent_permission = true
+    elsif parent_permission == 'N' || parent_permission.downcase == 'n'
+      parent_permission = false
+    else
+      return nil
+    end
+
+    Student.new(age, classroom, parent_permission, name)
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    Teacher.new(age, specialization, name)
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age} "
+    end
+
+    person_index = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals_by_person_id
+    print 'ID of person: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'Rentals:'
+    rentals.each { |rental| puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" }
+  end
+end
+
+def main
+  app = App.new
+
+  puts 'Welcome to School Library App!'
+
+  options = {
+    'List all books' => :list_books,
+    'List all people' => :list_people,
+    'Create a person' => :create_person,
+    'Create a book' => :create_book,
+    'Create a rental' => :create_rental,
+    'List all rentals for a given person id' => :list_rentals_by_person_id,
+    'Exit' => :exit
+  }
+
+  loop_method(app, options)
+end
+
+def loop_method(app, options)
+  loop do
+    option = get_menu_option(options)
+    method = options[option]
+    if method == :exit
+      puts 'Thank you for using this app!'
+      break
+    else
+      app.send(method)
+    end
+
+    puts 'Press any key to continue'
+    gets.chomp
+  end
+end
+
+def get_menu_option(options)
+  puts 'Please choose an option by entering a number:'
+  options.each_with_index { |(option, _), index| puts "#{index}) #{option}" }
```

Comment:
- [ ] Kindly make sure to add a `main.rb` file which will work as the entry point as stated in the requirements.

<img width="819" alt="Screenshot 2022-12-22 at 2 38 50 PM" src="https://user-images.githubusercontent.com/70269507/209128071-a023baa1-2098-48ed-ab99-ef4dbac45a7f.png">